libs/core/include/core/byte_array/consumers.hpp

```diff
@@ -192,6 +192,52 @@ int StringConsumer(byte_array::ConstByteArray const &str, uint64_t &pos)
   return STRING;
 }
 
+template <int STRING>
```

Comment:
This pattern looks a bit oddly inefficient - could you not pass as a parameter an int to return on success

Reply:
The main reason for this pattern is that consumers have a certain function signature that is ignorant to the return value. Another pattern that could be followed would be
```
ConsumerFunction StringConsumerFactory(int type)
{
  return [type](byte_array::ConstByteArray const &str, uint64_t &pos, char quoteChar)
  {
     // ...
     return type;
   }
}
```